### PR TITLE
Update libraries (cats, cats-effect, cats-mtl, scalatest), drop Scala 2.11 support

### PR DESCRIPTION
**Pull request by iRevive**
- Drop Scala 2.11 support
- Update scalajs to 1.2.0
- Update cats, cats-effect to 2.2.0
- Update cats-mtl to 1.0.0

### CODE REVIEW

**Comment by iRevive** [comment]
Seems I need to update mima rules. Can someone assist me with the place where I should start?  
I can assume I should exclude old artifacts from the verifications.

**Comment by SethTisue** [comment]
I'm interested in seeing this merged because I want it for https://github.com/scala/community-build/pull/1231

**Comment by cb372** [comment]
Sorry, missed this! I'll take a look

**Review by cb372 — COMMENTED** [review]
These changes LGTM.

Regarding bincompat, we'll definitely need to bump the version to 2.0.0 for the MTL module at least, because there are lots of breaking changes there. I think it's simplest to bump the version globally rather than maintaining a separate versioning scheme just for that module.

That means we don't need to worry about bincompat, so we can just disable mima until the next release. We are due a release, so we can release v2.0.0 as soon as this is merged.

You can disable the mima check by setting `mimaPreviousArtifacts := Set.empty` for all modules.

@LukaJCB Does this sound reasonable to you?

File: modules/core/shared/src/test/scala/retry/PackageObjectSpec.scala [hunk 1]
```diff
@@ -94,7 +83,7 @@ class PackageObjectSpec extends AnyFlatSpec {
   it should "retry until the action succeeds" in new TestContext {
     val policy = RetryPolicies.constantDelay[StringOr](1.second)
 
-    val finalResult = retryingOnSomeErrors(
+    val finalResult = retryingOnSomeErrors[String].apply[StringOr, String](
```
[the remark this answers is not in the document]
Comment (iRevive) [reply modules/core/shared/src/test/scala/retry/PackageObjectSpec.scala:86]: @cb372 unfortunately, it was an inference problem

[the remark this answers is not in the document]
Comment (LukaJCB) [reply modules/core/shared/src/test/scala/retry/PackageObjectSpec.scala:86]: This is odd to me, I could understand if we introduced type inference problems in the mtl combinators, but this seems to be the standard one and I don't understand how this one changed to create this problem. I definitely think we should fix this before release.

**Review by LukaJCB — COMMENTED** [review]
Thanks a ton for taking the time to contribute this, I'm a bit worried about the type inference issue, but hopefully we can resolve that before making a release

**Comment by iRevive** [comment]
Thank you for the review.
I can investigate an issue with type inference on Sunday.

**Comment by iRevive** [comment]
@cb372 @LukaJCB turned out Intellij fooled me this time. There is no inference problem, the project compiles successfully with SBT. I disabled the mima checks as well.

**Review by cb372 — APPROVED** [review]
Thanks for investigating the type inference issue. IntelliJ strikes again 😄

**Comment by LukaJCB** [comment]
Awesome thanks for your work @iRevive!

**Pull request merged**